pkg/printer/printer.go

```diff
@@ -23,6 +24,7 @@ type WarningInfo struct {
 	Caption     string
 	Occurrences int
 	Suggestion  string
+	Matches     []*cliClient.Match
```

Comment:
I think `rule` object should not be aware of the concept of `matches`.
This is just the data type from the backend, but should not be part of the final object. I think saving a array of specific identifiers -> `metadataName` and `kind` is more accurate.

Matches includes rule-to-file mapping, while we are reversing the order of the print to file-to-rule mapping